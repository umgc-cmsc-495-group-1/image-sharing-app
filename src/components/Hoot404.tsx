import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Button, Container, Typography } from '@mui/material';
import { Box } from '@mui/system';

export default function Hoot404() {
    const navigate = useNavigate();
    return (
        <Container component="main" maxWidth="md" data-testid="hoot-404-container">
            <Box sx={{
                marginTop: 8,
                padding: 5,
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
            }}>
                <Typography variant='h2'>Uh Oh! It looks like there is nothing here...</Typography>
                <Button
                    variant="contained"
                    color="primary"
                    onClick={() => navigate(-1)}
                    data-testid="hoot-404-button"
                    sx={{
                        marginTop: 5,
                        padding: '0, 1.5rem. 0 1.5rem',
                        fontSize: '1.5rem'
                    }}
                >Go Back</Button>
            </Box>
        </Container>
    )
}
