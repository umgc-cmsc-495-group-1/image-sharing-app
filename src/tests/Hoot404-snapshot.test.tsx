import React from 'react';
import "@testing-library/react/dont-cleanup-after-each";
import '@testing-library/jest-dom';
import { 
  MemoryRouter,
  Routes,
  Route
} from 'react-router-dom';
import Hoot404  from '../components/Hoot404';
import { create } from 'react-test-renderer';

function renderWithMemoryRouter(component: JSX.Element) {
  return create(
    <MemoryRouter>
      <Routes>
        <Route path='*' element={component} />
      </Routes>
    </MemoryRouter>
  );
}
it('Renders correctly',() => {
  const renderer = renderWithMemoryRouter(<Hoot404 />);
  expect(renderer.toJSON()).toMatchSnapshot();
})

