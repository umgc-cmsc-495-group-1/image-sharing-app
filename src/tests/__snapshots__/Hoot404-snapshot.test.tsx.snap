// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  data-testid="hoot-404-container"
>
  <div
    className="MuiBox-root css-1jn8zvf"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
    >
      Uh Oh! It looks like there is nothing here...
    </h2>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-17bgigr-MuiButtonBase-root-MuiButton-root"
      data-testid="hoot-404-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Go Back
    </button>
  </div>
</main>
`;
